### ISSUE TITLE
Bracket and intercept do not work together

### ISSUE THREAD

**Issue by rtimush** [polysemy-research/polysemy#133]
It seems that effects decorated with an `intercept` combinator are ignored when called inside a bracket. For example, the following code prints
```
writeTTY "Without bracket"
Without bracket
With bracket
```
although I expected to have a  `writeTTY "With bracket"` line as well.

```haskell
import           Polysemy
import           Polysemy.Resource

data Teletype m a where
    WriteTTY :: String -> Teletype m ()

makeSem ''Teletype

runTeletypeIO :: Member (Lift IO) r => Sem (Teletype ': r) a -> Sem r a
runTeletypeIO = interpret $ \case
    WriteTTY msg -> sendM $ putStrLn msg

withTeletypeLogging :: Members '[Teletype] r => Sem r a -> Sem r a
withTeletypeLogging = intercept $ \case
    WriteTTY msg -> do
        writeTTY $ "writeTTY " <> show msg
        writeTTY msg

main :: IO ()
main = (runM .@ runResourceInIO) . runTeletypeIO . withTeletypeLogging $ do
    writeTTY "Without bracket"
    bracket (return ()) return (const $ writeTTY "With bracket")
```

**Comment by isovector** [comment]
Weird. Mind trying this in a `Polysemy.Error.catch` and see if you can repro it? In the three cases of:

1. `runError . runTeletypeIO . withTeletypeLogging`
2. `runTeletypeIO . withTeletypeLogging . runError`
3. `runTeletypeIO . runError . withTeletypeLogging`

**Comment by rtimush** [comment]
Sure, tried with `catch` — works for (2), doesn't work for (1) and (3).
Same with `bracket` — if I use `runResource` and put it after `withTeletypeLogging` it also starts working.

**Comment by isovector** [comment]
That's slightly encouraging. I'm not sure what the heck is going on, but at least it's consistent with the rest of the semantics. Let me dig in.

**Comment by isovector** [comment]
Oh, yeah. This makes sense.

```haskell
interceptH f (Sem m) = Sem $ \k -> m $ \u ->
  case prj u of
    Just (Yo e s d y v) ->
      usingSem k $ fmap y $ runTactics s (raise . d) v $ f e
    Nothing -> k u
```

The interpretation of `interpretH` doesn't do any weaving, so obviously it's not going to get installed in the sub-computations.

This is a total bug --- thanks for catching it!

**Comment by isovector** [comment]
Fixed in [0.5.0.1](https://hackage.haskell.org/package/polysemy-0.5.0.1). Thanks again for the report!

**Comment by rtimush** [comment]
Thanks for a super quick fix!
